src/test/java/com/epam/aidial/cfg/service/config/syncstate/EntitySyncStateResolverTest.java

```diff
@@ -0,0 +1,181 @@
+
+        CoreConfigReloadCache.Entry cacheEntry = new CoreConfigReloadCache.Entry(config, configReloadTimestamp);
+        when(coreConfigReloadCache.get()).thenReturn(cacheEntry);
+
+        when(objectMapper.valueToTree(currentState)).thenReturn(currentStateJsonNode);
+
+        when(syncStateStatusResolver.resolve(currentStateJsonNode, null, true, currentStateUpdatedAt, configReloadTimestamp))
+                .thenReturn(EntitySyncStateStatus.IN_PROGRESS);
+
+        EntitySyncState expectedResult = new EntitySyncState(currentStateJsonNode, null, EntitySyncStateStatus.IN_PROGRESS);
+
+        // when
+        EntitySyncState actualResult = syncStateResolver.resolve(currentState, true, currentStateUpdatedAt, "entities", "entityName");
+
+        // then
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void resolve_configHasRequestedEntitiesAndHasEntityWithRequestedKey_returnsFullState() {
+        // given
+        long currentStateUpdatedAt = 1;
+        long configReloadTimestamp = 2;
+
+        CurrentState currentState = new CurrentState();
+        ObjectNode currentStateJsonNode = JsonNodeFactory.instance.objectNode();
+
+        ObjectNode requestedEntity = JsonNodeFactory.instance.objectNode();
+        requestedEntity.put("prop1", "prop1Value");
+        requestedEntity.put("prop2", "prop2Value");
+
+        ObjectNode requestedEntities = JsonNodeFactory.instance.objectNode();
+        requestedEntities.set("entityName", requestedEntity);
+
+        ObjectNode config = JsonNodeFactory.instance.objectNode();
+        config.set("entities", requestedEntities);
+
+        CoreConfigReloadCache.Entry cacheEntry = new CoreConfigReloadCache.Entry(config, configReloadTimestamp);
+        when(coreConfigReloadCache.get()).thenReturn(cacheEntry);
+
+        when(objectMapper.valueToTree(currentState)).thenReturn(currentStateJsonNode);
+
+        when(syncStateStatusResolver.resolve(currentStateJsonNode, requestedEntity, true, currentStateUpdatedAt, configReloadTimestamp))
+                .thenReturn(EntitySyncStateStatus.FULLY_SYNCED);
+
+        EntitySyncState expectedResult = new EntitySyncState(currentStateJsonNode, requestedEntity, EntitySyncStateStatus.FULLY_SYNCED);
+
+        // when
+        EntitySyncState actualResult = syncStateResolver.resolve(currentState, true, currentStateUpdatedAt, "entities", "entityName");
+
+        // then
+        assertThat(actualResult).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void resolve_configHasRequestedEntitiesAndHasEntityWithRequestedKeyAndCurrentStateIsInvalid_returnsStateWithoutCurrentStatePart() {
+        // given
+        long currentStateUpdatedAt = 1;
+        long configReloadTimestamp = 2;
+
+        CurrentState currentState = new CurrentState();
+        ObjectNode currentStateJsonNode = JsonNodeFactory.instance.objectNode();
+
+        ObjectNode requestedEntity = JsonNodeFactory.instance.objectNode();
```

Comment:
it's hard to read this JsonNodeFactory chains. may be better to define json as json string and parse it in spec method.